lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -711,6 +725,25 @@ public void run() {
 		}
 	}
 
+	private void handleReauth() {
+		pendingReauth = false;
```

Comment:
I'm not sure how many attempts should be given for reauth. If we don't clear it the thread will exit `tryWait()` immediately and will consume all the CPU time if `authCallback` keeps failing.